services/worker/src/worker/job_runners/split/descriptive_statistics.py

```diff
@@ -503,44 +514,41 @@ def compute_descriptive_statistics_response(
 
     logging.info(f"compute 'split-descriptive-statistics' for {dataset=} {config=} {split=}")
 
-    config_parquet_and_info_step = "config-parquet-and-info"
-    parquet_and_info_best_response = get_previous_step_or_raise(
-        kinds=[config_parquet_and_info_step],
+    # get parquet urls and dataset_info
+    config_parquet_metadata_step = "config-parquet-metadata"
+    parquet_metadata_best_response = get_previous_step_or_raise(
+        kinds=[config_parquet_metadata_step],
         dataset=dataset,
         config=config,
     )
-    content_parquet_and_info = parquet_and_info_best_response.response["content"]
+    content_parquet_metadata = parquet_metadata_best_response.response["content"]
     try:
         split_parquet_files = [
             parquet_file
-            for parquet_file in content_parquet_and_info["parquet_files"]
+            for parquet_file in content_parquet_metadata["parquet_files_metadata"]
             if parquet_file["config"] == config and parquet_file["split"] == split
         ]
-        dataset_info = content_parquet_and_info["dataset_info"]
+        features = content_parquet_metadata["features"]
+
     except KeyError as e:
         raise PreviousStepFormatError(
-            (
-                f"Previous step '{config_parquet_and_info_step}' did not return the expected content: "
-                "'parquet_files' or 'dataset_info'. "
-            ),
-            e,
+            f"Previous step '{config_parquet_metadata_step}' did not return the expected content", e
         ) from e
 
     if not split_parquet_files:
         raise ParquetResponseEmptyError("No parquet files found.")
-    features = dataset_info.get("features")
-    if features is None:
-        raise PreviousStepFormatError(
-            f"Previous step '{config_parquet_and_info_step}' did not return the expected content: "
-            "no features found in 'dataset_info'. "
-        )
 
-    split_parquets_size = sum(parquet_file["size"] for parquet_file in split_parquet_files)
-    if split_parquets_size > max_parquet_size_bytes:
-        raise SplitWithTooBigParquetError(
-            f"Statistics computation is limited to split parquets under {max_parquet_size_bytes} bytes. "
-            f"Current size of sum of split parquets is {split_parquets_size} bytes."
-        )
+    if not features:
+        raise FeaturesError("No features found in cache. ")
```

Comment:
not sure about this error, maybe better `PreviousStepFormatError`?

Reply:
I'm not sure it can exist. But I would say: create a new exception (`EmptyFeaturesError`?) so that we know where it comes from in case it raises.

Reply:
Or do we ensure in config-parquet-and-info that it's never empty? In that case, yes, it would be `PreviousStepFormatError` I think

Reply:
Hmmm: we don't at https://github.com/huggingface/datasets-server/blob/42d945dc5702d1a3016a699993078bb01fdf6774/services/worker/src/worker/job_runners/config/parquet_and_info.py#L1260

So: it's theorically possible for the features dict to be empty